chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -1034,12 +1038,14 @@ public synchronized void loadChunks(TaskTracker taskTracker, World world, Collec
         int wx0 = cp.x * 16; // Start of this chunk in world coordinates.
         int wz0 = cp.z * 16;
         BiomeData biomeData = chunkData.getBiomeData();
-        for (int cz = 0; cz < 16; ++cz) {
-          int wz = cz + wz0;
-          for (int cx = 0; cx < 16; ++cx) {
-            int wx = cx + wx0;
-            int biomePaletteIdx = biomeData.getBiome(cx, 0, cz); // TODO add vertical biomes support (1.15+)
-            biomePaletteIdxMap.set(biomePaletteIdx, wx, wz);
+        for (int y = chunkData.minY(); y < chunkData.maxY(); y++) {
+          for (int cz = 0; cz < 16; ++cz) {
+            int wz = cz + wz0;
+            for (int cx = 0; cx < 16; ++cx) {
+              int wx = cx + wx0;
+              int biomePaletteIdx = biomeData.getBiome(cx, y, cz); // TODO add vertical biomes support (1.15+)
```

Comment:
This TODO is no longer relevant as this is what you've just done or is there still something to do?

Reply:
good point - yeah this can be removed. I'll mark this pr as draft for now as there are a lot of silly things around